### PR TITLE
[CI-only] Use pattern matching for release_branches

### PR DESCRIPTION
**Pull request by mdeggies**
### Description
Pattern matching was [recently added](https://github.com/hashicorp/crt-orchestrator/pull/51) so that teams no longer have to explicitly list every branch that should trigger the CRT pipeline. This simplifies release preparation- anytime a new release branch is created, it will produce releasable artifacts and exercise the full pipeline.

### Testing & Reproduction steps
This has been tested in multiple projects since being rolled out. There are no nomad-specific tests that need to be done.

### Links
PR where this functionality was added: https://github.com/hashicorp/crt-orchestrator/pull/51

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [X] not a security concern

### CODE REVIEW

File: .release/ci.hcl [hunk 1]
```diff
@@ -13,10 +13,7 @@ project "nomad" {
 
     release_branches = [
       "main",
-      "release/1.0.x",
-      "release/1.1.x",
-      "release/1.2.x",
-      "release/1.3.x",
+      "release/**",
```
Comment (tgross) [inline .release/ci.hcl:16]: Won't this trigger release workflows on dead branches? Ex. https://github.com/hashicorp/nomad/branches/all?query=release%2F

Comment (mdeggies) [reply .release/ci.hcl:16]: The pipeline won't be exercised until a new commit is made to the branch. When I say pipeline, I mean all of the common workflows that run after a commit is made on a release branch (not including the promotions to staging/production, which still require a manual trigger using `bob`).

**Comment by mdeggies** [comment]
Can I get an approval please? Thanks!

**Review by tgross — APPROVED** [review]
LGTM!

**Review by kpenfound — APPROVED** [review]
🚀

**Pull request merged**